src/plugins/data_table/dataTables.colReorder.js

```diff
@@ -30,1901 +30,1877 @@ import {Coordinates, DomVisibility} from "../../utils/dom_visibility";
-        iOrigColumn = oSettings.aoColumns[i]._ColReorder_iOrigCol;
+      /**
+       * Because we change the indexes of columns in the table, relative to their starting point
+       * we need to reorder the state columns to what they are at the starting point so we can
+       * then rearrange them again on state load!
+       *  @method  _fnStateSave
+       *  @param   object oState DataTables state
+       *  @returns string JSON encoded cookie string for DataTables
+       *  @private
+       */
+      "_fnStateSave": function (oState) {
+        var i, iLen, aCopy, iOrigColumn;
+        var oSettings = this.s.dt;
 
-        /* Column filter */
-        oState.aoSearchCols[iOrigColumn] = aSearchCopy[i];
+        /* Sorting */
+        for (i = 0; i < oState.aaSorting.length; i++) {
+          oState.aaSorting[i][0] = oSettings.aoColumns[oState.aaSorting[i][0]].colReorderIOrigCol;
+        }
 
-        /* Visibility */
-        oState.abVisCols[iOrigColumn] = oSettings.aoColumns[i].bVisible;
+        var aSearchCopy = $.extend(true, [], oState.aoSearchCols);
+        oState.ColReorder = [];
+        oState.ColSizes = [];
 
-        /* Column reordering */
-        oState.ColReorder.push(iOrigColumn);
+        for (i = 0, iLen = oSettings.aoColumns.length; i < iLen; i++) {
+          iOrigColumn = oSettings.aoColumns[i].colReorderIOrigCol;
 
-        /* Column Sizes */
-        oState.ColSizes[iOrigColumn] = oSettings.aoColumns[i].sWidth;
-      }
-    },
+          /* Column filter */
+          oState.aoSearchCols[iOrigColumn] = aSearchCopy[i];
 
+          /* Visibility */
+          oState.abVisCols[iOrigColumn] = oSettings.aoColumns[i].bVisible;
 
-    /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
-     * Mouse drop and drag
-     */
+          /* Column reordering */
+          oState.ColReorder.push(iOrigColumn);
 
-    /**
-     * Add a mouse down listener to a particluar TH element
-     *  @method  _fnMouseListener
-     *  @param   int i Column index
-     *  @param   element nTh TH element clicked on
-     *  @returns void
-     *  @private
-     */
-    "_fnMouseListener": function (i, nTh) {
-      var that = this;
-
-      var thead = $(nTh).closest('thead');
-      // listen to mousemove event for resize
-      if (this.s.allowResize) {
-        //$(nTh).bind('mousemove.ColReorder', function (e) {
-        thead.bind('mousemove.ColReorder', function (e) {
-          var nTable = that.s.dt.nTable;
-          if (that.dom.drag === null && that.dom.resize === null) {
-            /* Store information about the mouse position */
-            var nThTarget = e.target.nodeName === "TH" ? e.target : $(e.target).parents('TH')[0];
-            var offset = $(nThTarget).offset();
-            var nLength = $(nThTarget).innerWidth();
-
-            /* are we on the col border (if so, resize col) */
-            if (Math.abs(e.pageX - Math.round(of
```

Comment:
Could improve readability by using a descriptively-named local variable.

Reply:
I'm going to leave it as is for now

Reply:
👍